### ISSUE TITLE
Discuss pros/cons - yaml versus helm charts versus cdk8

### ISSUE THREAD

**Issue by mkbhanda** [opea-project/GenAIInfra#65]
Currently we are deploying yaml based definition of microservices. They are easy to create and use. Helm charts are more complex but they allow modifying the template, passing in HW information to even determine which image to download. CDK8 is yet another format that allows varying allocated resources. With Service mesh both yaml and helm charts work to trigger launch of services. 
It would be good to pause and determine if we should support one or more formats of each for the sample microservices and GMC support connecting services in any format.

**Comment by eero-t** [comment]
FYI: Helm (explicitly) does **not** support updating nor removing CRDs, for those one needs to apply/remove manifests manually any way.

If common configuration options are in e.g shared configMaps, instead of hard-coded to multiple pod specs, IMHO editing them is not much harder than using Helm.  IMHO more interesting question is which of those options could be updated at runtime / do OPEA services have config reloader like e.g. Prometheus services have?

**Comment by mkbhanda** [comment]
Thank you @eero-t ! Will be leveraging your expertise in this space.

**Comment by jfding** [comment]
@mkbhanda We have already nailed down the decisions, so it looks like we can close this issue now.

**Comment by mkbhanda** [comment]
Yes, we have adopted Helm charts and now that we are not using or maintaining GMC where we used yaml files, we can close this issue.
Thanks @jfding and @eero-t !
